examples/asset/alter_mesh.rs

```diff
@@ -0,0 +1,200 @@
+        },
+    ));
+}
+
+fn instructions(mut commands: Commands) {
+    commands
+        .spawn((
+            Name::new("Instructions"),
+            NodeBundle {
+                style: Style {
+                    align_items: AlignItems::Start,
+                    flex_direction: FlexDirection::Column,
+                    justify_content: JustifyContent::Start,
+                    width: Val::Percent(100.),
+                    ..default()
+                },
+                ..default()
+            },
+        ))
+        .with_children(|parent| {
+            parent.spawn(TextBundle::from_section(
+                "Space: swap meshes by mutating a Handle<Mesh>",
+                TextStyle::default(),
+            ));
+            parent.spawn(TextBundle::from_section(
+                "Return: mutate the mesh itself, changing all copies of it",
+                TextStyle::default(),
+            ));
+        });
+}
+
+#[derive(Default)]
+struct IsMeshScaled(bool);
+
+fn keyboard_controls(
+    asset_server: Res<AssetServer>,
+    mut is_mesh_scaled: Local<IsMeshScaled>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    keyboard_input: Res<ButtonInput<KeyCode>>,
+    left_shape: Query<&Handle<Mesh>, With<Left>>,
+    mut right_shape: Query<(&mut Handle<Mesh>, &mut Shape), Without<Left>>,
+) {
+    if keyboard_input.just_pressed(KeyCode::Space) {
+        // Mesh handles, like other parts of the ECS, can be queried as mutable and modified at
+        // runtime. We only spawned one shape without the `Left` marker component.
+        let Ok((mut handle, mut shape)) = right_shape.get_single_mut() else {
+            return;
+        };
+
+        // Switch to a new Shape variant
+        shape.set_next_variant();
+
+        // Modify the handle associated with the Shape on the right side. Note that we will only
+        // have to load the same path from storage media once: repeated attempts will re-use the
+        // asset.
+        *handle = asset_server.load(
+            GltfAssetLabel::Primitive {
+                mesh: 0,
+                primitive: 0,
+            }
+            .from_asset(shape.get_model_path()),
+        );
+    }
+
+    if keyboard_input.just_pressed(KeyCode::Enter) {
+        // It's convenient to retrieve the asset handle stored with the shape on the left. However,
+        // we could just as easily have retained this in a resource or a dedicated component.
+        let Ok(handle) = left_shape.get_single() else {
+            return;
+        };
+
+        // Obtain a mutable reference to the Mesh asset.
+        let Some(mesh) = meshes.get_mut(handle) else {
+            return;
+        };
+
+        // Now we can directly manipulate vertices on the mesh. Here, we're just scaling in and out
+        // for demonstration purposes. This will affect all entities currently using the asset.
+        if let Some(VertexAttributeValues::Float32x3(positions)) =
+            mesh.attribute_mut(Mesh::ATTRIBUTE_POSITION)
```

Comment:
I will freely admit this took quite a lot of getting to a "working" state, so if there's a more sensible way to do it let me know! Much head-scratching involved :laughing:

Reply:
Yep, our mesh modification API is not particularly ergonomic: it's a very rare operation to do in code!

Reply:
Pretty sure this this is the only way to do this, yeah

Reply:
Pretty sure this this is the only way to do this, yeah